util/jemalloc_nodump_allocator.cc

```diff
@@ -48,12 +120,41 @@ void* JemallocNodumpAllocator::Alloc(extent_hooks_t* extent, void* new_addr,
   return result;
 }
 
-JemallocNodumpAllocator::~JemallocNodumpAllocator() {
-  assert(arena_index_ != 0);
-  std::string key = "arena." + ToString(arena_index_) + ".destroy";
+Status JemallocNodumpAllocator::DestroyArena(unsigned arena_index) {
+  assert(arena_index != 0);
+  std::string key = "arena." + ToString(arena_index) + ".destroy";
   int ret = mallctl(key.c_str(), nullptr, 0, nullptr, 0);
   if (ret != 0) {
-    fprintf(stderr, "Failed to destroy jemalloc arena, error code: %d\n", ret);
+    return Status::Incomplete("Failed to destroy jemalloc arena, error code: " +
+                              ToString(ret));
+  }
+  return Status::OK();
+}
+
+void JemallocNodumpAllocator::DestroyThreadSpecificCache(void* ptr) {
```

Comment:
This needs to be hooked into thread cleanup handlers.  We may never destroy the allocator, but may have new threads come in and old thread die all the time. If we don't release the tcache, it becomes a leak.

Reply:
yes, the `ThreadLocalPtr` object handles thread exit and call the handler: https://github.com/facebook/rocksdb/blob/master/util/thread_local.cc#L279
In the constructor we set the handler to `DestroyThreadSpecificCache`.